lib/chore/job.rb

```diff
@@ -108,6 +114,25 @@ def prefixed_queue_name
       def has_backoff?
         self.options.key?(:backoff)
       end
+
+      def has_dedupe_lambda?
+        self.options.key?(:dedupe_lambda)
+      end
+
+      def dedupe_key(*args)
+        return unless has_dedupe_lambda?
+        if self.options[:dedupe_lambda].arity != self.instance_method(:perform).arity
+          raise ArgumentError, "#{self.to_s}: dedupe_lambda must accept the same number of arguments as perform"
+        end
+        # run the proc to get the key
+        begin
+          key = self.options[:dedupe_lambda].call(*args)
+        rescue => e
+          raise "#{self.to_s}: dedupe_lambda raised error: #{e}"
```

Comment:
We lose stack trace and exception class name information when we re-raise like this.  This would end up reporting to NewRelic as StandardError, which could end up getting combined with other StandardErrors.  You also would only get a stack trace starting at this line.

I think we could get away without a begin/rescue here unless we actually want the error condition to change something about the behavior of the dedupe.